app/uk/gov/hmrc/gform/views/summary/snippets/hmrc_tax_period.scala.html

```diff
@@ -0,0 +1,48 @@
+@*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.gform.sharedmodel.form._
+@import uk.gov.hmrc.gform.sharedmodel.formtemplate._
+@import uk.gov.hmrc.gform.validation.{FormFieldValidationResult, ComponentField}
+@import uk.gov.hmrc.gform.views._
+@import uk.gov.hmrc.gform.views.html.localisation
+@import uk.gov.hmrc.gform.views.summary.TextFormatter
+@import uk.gov.hmrc.play.views.html.helpers._
+@import uk.gov.hmrc.gform.models.helpers.TaxPeriodHelper._
+
+@(fieldValue: FormComponent, validationResult: Option[FormFieldValidationResult], changeButton: Html)
+
+
+@hasErrors = @{validationResult.exists(_.isNotOk)}
+
+@errors = @{validationResult.map(_.fieldErrors.toList).getOrElse(Set().toList).map{ e =>
+    errorInline("summary", e, Seq()) }}
+
+@formatTaxPeriod(taxPeriod:String) = @{
+val l =taxPeriod.split('|')
+localisation("From")+" "+l(0)+" "+localisation("to")+" "+l(1)
```

Comment:
Parsing taxPeriod is unsafe. Why don't do something like this:

```
taxPeriod.split('|').toList match {
  case from :: to :: _ => localisation("From")+" "+from+" "+localisation("to")+" "+to
  case _ => ??? // Too late to react on missing pipe
}
```

But it still isn't perfect as we cannot react on wrong data. This should be done much earlier and `OptionParams` (I guess) object should be pass here.